ui/server.py

```diff
@@ -407,6 +408,22 @@ def update_discovery_group():
         return 'OK', 200
 
 
+@app.route('/update_similar_discoveries', methods=['POST'])
+def update_similar_discoveries():
+    target_snippet = request.form.get('snippet')
+    state = request.form.get('state')
+    url = request.form.get('url')
+    file = request.form.get('file')
+
+    response1 = c.update_discovery_group(state, url, file, target_snippet)
+    response2 = c.update_similar_snippets(target_snippet,
+                                          state,
+                                          url)
+    if response1 is False or response2 is False:
+        return 'Error in updating similar snippets', 500
```

Comment:
can this happen? 
According to the `if`, you may update the similar ones without updating the "discovery group"

Reply:
shouldn't we do something like
```
response1 = c.update_discovery_group(state, url, file, target_snippet)

if not response1:
    return '...', 500
response2 = c.update_similar_snippets(target_snippet,state,url)
if not response2:
    return '...', 500
return '...', 200
```

Reply:
yes it's better, I thought about it